server/bin/www

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env node
+
+/**
+ * Module dependencies.
+ */
+var app = require('../app');
+var debug = require('debug');
+var http = require('http');
+var amqp = require('amqplib/callback_api');
+var io = require('socket.io');
+
+/**
+ * Get port from environment and store in Express.
+ */
+var port = normalizePort(process.env.PORT || 4000);
+app.set('port', port);
+
+/**
+ * Create HTTP server.
+ */
+var server = http.createServer(app);
+
+/**
+ * Listen on provided port, on all network interfaces.
+ */
+server.listen(port);
+server.on('error', onError);
+server.on('listening', onListening);
+
+/* Setup socket.io stuff for live webpage */
+io = io.listen(server);
+
+/* Setup the AMQP channel with RabbitMQ (change host if needed) */
+var rmqport = '5672';
+amqp.connect('amqp://localhost:' + rmqport, function(err, conn) {
+  try {
+    conn.createChannel(function(err, ch) {
+      // NOTE: the string in this variable should be 
+      // the same in the provider
+      var q = 'msgQueue';
+      var ex = 'hermesExchange';
+  
+      // check if queue is valid and wait for messages
+      ch.assertExchange(ex, 'fanout', {durable: false});
+      ch.assertQueue(q, {durable: false});
+      ch.bindQueue(q.queue, ex, '');
+      console.log("Express: waiting for messages in %s", q);
+      ch.consume(q, function(msg) {
+        var msgString = JSON.stringify(msg);
+        // update webpage at index
+        io.sockets.emit('add data', {
+          data: msgString
```

Comment:
So, `{data: msgString}` creates an object with an attribute `data` that has a value of `msgString`, so in `index.pug`

```
    socket.on('add data', function(data) {
      addListItem(data);
    });

    function addListItem(data) {z
      var dataText = document.createTextNode(data);
      var listItem = document.createElement('li');
      var dataList = document.getElementById('dataList');

      listItem.appendChild(dataText);
      dataList.appendChild(listItem);
    }

``` 
An object of the type `data` we specified above, called `data`, is getting added to the textNode. So, that's why it was printing out an object! 

Some alternatives, 
in `index.pug`
`var dataText = document.createTextNode(data.data);` <- would definitely change the name of that.
or here:
`io.sockets.emit('add data', msgString`) <- not sure if that's the best practice